service/src/main/java/org/databiosphere/workspacedataservice/workspacemanager/WorkspaceManagerDao.java

```diff
@@ -81,37 +90,39 @@ public ResourceList enumerateDataRepoSnapshotReferences(UUID workspaceId, int of
 
   /** Retrieves the azure storage container url and sas token for a given workspace. */
   public String getBlobStorageUrl(String storageWorkspaceId, String authToken) {
-    final ControlledAzureResourceApi azureResourceApi =
-        this.workspaceManagerClientFactory.getAzureResourceApi(authToken);
-    int count = 0;
-    int maxTries = 3;
-    while (true) {
-      try {
-        UUID workspaceUUID = UUID.fromString(storageWorkspaceId);
+    try {
+      final ControlledAzureResourceApi azureResourceApi =
+          this.workspaceManagerClientFactory.getAzureResourceApi(authToken);
+      UUID workspaceUUID = UUID.fromString(storageWorkspaceId);
+      LOGGER.debug(
+          "Finding storage resource for workspace {} from Workspace Manager ...", workspaceUUID);
+      ResourceList resourceList =
+          enumerateResources(
+              workspaceUUID, 0, 5, ResourceType.AZURE_STORAGE_CONTAINER, null, authToken);
+      // note: it is possible a workspace may have more than one storage container associated with
+      // it
+      // but currently there is no way to tell which one is the primary except for checking the
+      // actual container name
+      var storageUUID = extractResourceId(resourceList, storageWorkspaceId);
+      if (storageUUID != null) {
         LOGGER.debug(
-            "Finding storage resource for workspace {} from Workspace Manager ...", workspaceUUID);
-        ResourceList resourceList =
-            enumerateResources(
-                workspaceUUID, 0, 5, ResourceType.AZURE_STORAGE_CONTAINER, null, authToken);
-        // note: it is possible a workspace may have more than one storage container associated with
-        // it
-        // but currently there is no way to tell which one is the primary except for checking the
-        // actual container name
-        var storageUUID = extractResourceId(resourceList, storageWorkspaceId);
-        if (storageUUID != null) {
-          LOGGER.debug(
-              "Requesting SAS token-enabled storage url or workspace {} from Workspace Manager ...",
-              workspaceUUID);
-          CreatedAzureStorageContainerSasToken sasBundle =
-              azureResourceApi.createAzureStorageContainerSasToken(
-                  workspaceUUID, storageUUID, null, null, null, null);
-          return sasBundle.getUrl();
-        } else
-          throw new ApiException(
-              "WorkspaceManagerDao: Can't locate a storage resource matching workspace Id. ");
-      } catch (ApiException e) {
-        if (++count == maxTries) throw new WorkspaceManagerException(e);
+            "Requesting SAS token-enabled storage url or workspace {} from Workspace Manager ...",
+            workspaceUUID);
+        RestCall<CreatedAzureStorageContainerSasToken> sasBundleFunction =
+            () ->
+                azureResourceApi.createAzureStorageContainerSasToken(
+                    workspaceUUID, storageUUID, null, null, null, null);
```

Comment:
I know you're not newly introducing this LOC here, but it's tough to discern the meaning of all the naked `null` parameters here.  A helpful trick I like to use to make these hard-to-read param lists is to add some inline comments ilke so:

```suggestion
                    workspaceUUID, 
                    storageUUID,
                    /* sasIpRange=*/ null, 
                    /* sasExpirationDuration=*/ null,
                    /* sasPermissions= */ null,
                    /* sasBlobName= */ null);
```

This applies for any large param list whose type and/or meaning are hard to infer without an IDE giving you hints.

Reply:
Also just [linking](https://github.com/DataBiosphere/terra-workspace-data-service/pull/396/files#r1383367587) a counter example where the parameter names are self-documenting and where this kind of comment isn't necessary.